docs/reference/index.rst

```diff
@@ -69,6 +69,7 @@ Snapcraft has defined processes underlying its basic operation.
     components
     linters
     layouts
+    hooks
```

Comment:
@mr-cal Just confirming, do you think this is the optimal knowledge hierarchy?

Reply:
I think so.  Everything above it seems more important/fundamental.